ToDoApp/ToDoApp/ToDoService.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  ToDoDetail.swift
+//  ToDoApp
+//
+//  Created by Trong Hieu Vuong on 2022-11-29.
+//
+
+import Foundation
+
+enum TaskError: Error {
+    case unavailableTask
+}
+
+struct Task {
+    var id: String = ""
+    var name: String = ""
+    var deadline: Date = Date.now
+    var status: Bool = false
+    init(name: String, deadline: Date?) {
+        self.name = name
+        self.deadline = deadline!
+    }
+}
+
+protocol ToDoService {
+    func createTask(name: String, deadline: Date?) -> Task
+    func updateTask(todo: Task) throws
+    func taskForToday() -> [Task]
+}
+
+final class FeaturesToDo: ToDoService {
+    private var tasks: [Task] = []
+    // Create a task
+    func createTask(name: String, deadline: Date?) -> Task {
+        let oneTask = Task(name: name, deadline: deadline)
+        tasks.append(oneTask)
+        return oneTask
+    }
+    // Update the task
+    func updateTask(todo: Task) throws {
+        guard let index = tasks.firstIndex(where: {
+            $0.id == todo.id
+        })else {
+            throw TaskError.unavailableTask
+        }
+        tasks.insert(todo, at: index)
+    }
+    // Get list of tasks for today
+    func taskForToday() -> [Task] {
+        let date = Date.now
+        let todayList = tasks.filter({$0.deadline == date})
```

Comment:
[Swift Date type](https://developer.apple.com/documentation/foundation/date) represent a time interval relative to an absolute reference date in seconds. This means that when Date.now is returning its value it returned time interval in seconds from 1970 to the time it has been invoked. That's why comparing deadline date with now will not work. in order to check if date's day is equal to today you need to use Calendar and its method isDateInToday https://developer.apple.com/documentation/foundation/calendar/2293243-isdateintoday I suggest to create a protocol which will have the same signature for the isDayInTodayMethod, conform Calendar to this protocol and inject it as a dependency so you could test the ToDoService implementation later